### PR TITLE
Add Elastic search for optimize

### PR DESCRIPTION
**Pull request by adusumillipraveen**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/elasticsearch.tf [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    azurerm           = azurerm
+    azurerm.mgmt      = azurerm.mgmt
+    azurerm.aks-infra = azurerm.aks-infra
+  }
+  logAnalyticsId      = data.azurerm_log_analytics_workspace.log_analytics.workspace_id
+  logAnalyticsKey     = data.azurerm_log_analytics_workspace.log_analytics.primary_shared_key
+  dynatrace_instance  = var.dynatrace_instance
+  dynatrace_hostgroup = var.dynatrace_hostgroup
+  dynatrace_token     = data.azurerm_key_vault_secret.dynatrace_token.value
+  enable_logstash     = "No"
```
Comment (timja) [inline infrastructure/elasticsearch.tf:37]: use booleans?

File: infrastructure/elasticsearch.tf [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  enable_kibana       = "No"
+  alerts_email        = data.azurerm_key_vault_secret.alerts_email.value
+}
+
+locals {
+  // Vault name
+  vaultName = "${var.raw_product}-${var.env}"
```
Comment (timja) [inline infrastructure/elasticsearch.tf:44]: why raw_product?

File: infrastructure/perftest.tfvars [hunk 3]
```diff
@@ -0,0 +1,8 @@
+dataNodesAreMasterEligible = "Yes"
+vmDataNodeCount            = "2"
+vmSizeAllNodes             = "Standard_D2_v2"
+storageAccountType         = "Default"
+vmDataDiskCount            = "2"
+
+
+aks_infra_subscription_id = "8a07fdcd-6abd-48b3-ad88-ff737a4b9e3c"
```
Comment (timja) [inline infrastructure/perftest.tfvars:8]: should this come in from jenkins instead?

File: infrastructure/variables.tf [hunk 4]
```diff
@@ -13,19 +13,56 @@ variable "component" {}
...
+
+variable "dataNodesAreMasterEligible" {
+  type    = string
+  default = "Yes"
+}
+
+variable "vmDataNodeCount" {
+  description = "number of data nodes"
+  type        = string
+  default     = "1"
```
Comment (timja) [inline infrastructure/variables.tf:45]: number instead of string?

File: infrastructure/variables.tf [hunk 5]
```diff
@@ -13,19 +13,56 @@ variable "component" {}
...
+}
+
+variable "vmSizeAllNodes" {
+  description = "vm size for all the cluster nodes"
+  type        = string
+  default     = "Standard_D2_v2"
```
Comment (timja) [inline infrastructure/variables.tf:51]: have you reviewed if this is the right size based on it being a current generation, right sized, and in our standard skus if one fits?

File: infrastructure/variables.tf [hunk 6]
```diff
@@ -13,19 +13,56 @@ variable "component" {}
...
+variable "storageAccountType" {
+  description = "disk storage account type"
+  type        = string
+  default     = "Standard"
+}
+
+variable "vmDataDiskCount" {
+  description = "number of data node's disks"
+  type        = string
+  default     = "1"
```
Comment (timja) [inline infrastructure/variables.tf:63]: number?

File: infrastructure/elasticsearch.tf (hunk 1 above)
Comment (adusumillipraveen) [reply infrastructure/elasticsearch.tf:37]: This is propagated to upstream, don't want to transform anything in between.  https://github.com/elastic/azure-marketplace/blob/master/src/mainTemplate.json#L210

File: infrastructure/elasticsearch.tf (hunk 2 above)
Comment (adusumillipraveen) [reply infrastructure/elasticsearch.tf:44]: just copy paste, changing it.

File: infrastructure/elasticsearch.tf (hunk 1 above)
Comment (timja) [reply infrastructure/elasticsearch.tf:37]: the module should do that, you have a terraform API, not an ARM API.

Consumers shouldn't have to care about that

File: infrastructure/variables.tf [hunk 7]
```diff
@@ -1,31 +1,63 @@
...
+  type        = string
   description = "This is the ID of the Application you wish to give access to the Key Vault via the access policy"
 }
+
+variable "aks_subscription_id" {}
+
+variable "mgmt_subscription_id" {}
+
+variable "dataNodesAreMasterEligible" {
+  type    = string
```
Comment (timja) [inline infrastructure/variables.tf:33]: boolean as well?

File: infrastructure/aat.tfvars [hunk 8]
```diff
@@ -0,0 +1,6 @@
+dataNodesAreMasterEligible = "Yes"
+vmDataNodeCount            = 2
+storageAccountType         = "Default"
+vmDataDiskCount            = 2
+dynatrace_instance         = "yrk32651"
```
Comment (timja) [inline infrastructure/aat.tfvars:5]: bit yuck putting this here, probably fine for now, I think if dynatrace were to be spread across the infra then it should be in it's own module like the common tags module

File: infrastructure/variables.tf [hunk 9]
```diff
@@ -1,31 +1,63 @@
...
+variable "storageAccountType" {
+  description = "disk storage account type"
+  type        = string
+  default     = "Standard"
+}
+
+variable "vmDataDiskCount" {
+  description = "number of data node's disks"
+  type        = number
+  default     = 1
```
Comment (timja) [inline infrastructure/variables.tf:58]: why would you have more than 1? what's the point in this being a var

File: infrastructure/variables.tf [hunk 10]
```diff
@@ -1,31 +1,63 @@
...
+
+variable "vmSizeAllNodes" {
+  description = "vm size for all the cluster nodes"
+  type        = string
+  default     = "Standard_D2s_v3"
+}
+
+variable "storageAccountType" {
+  description = "disk storage account type"
+  type        = string
```
Comment (timja) [inline infrastructure/variables.tf:51]: string is the default type, no need to specify it

**Review by timja — APPROVED** [review]

**Comment by adusumillipraveen** [comment]
Moving this to shared infra

**Pull request closed without merging**